app/models/service_template_ansible_playbook.rb

```diff
@@ -138,22 +143,44 @@ def job_template(action)
     resource_actions.find_by!(:action => action.to_s.capitalize).configuration_template
   end
 
+  def new_dialogs(config_info)
+    [:provision, :retirement, :reconfigure].each do |action|
+      info = config_info[action]
+      if new_dialog_required(info)
+        info[:dialog_id] = create_new_dialog(info[:new_dialog_name], job_template(action), info[:hosts]).id
+      end
+    end
+  end
+
+  def new_job_templates(config_info, name, description, auth_user)
+    config_info.deep_merge!(send(:create_job_template, name, description, config_info, auth_user))
+  end
+
+  def job_template_modifications(name, description, config_info, auth_user)
+    [:provision, :retirement, :reconfigure].each do |action|
+      info = config_info[action]
+      if info && info.key?(:playbook_id) && !info.key?(:create_only)
+        tower, params = self.class.send(:build_parameter_list, "miq_#{name}_#{action}", description, info)
+        params[:manager_ref] = job_template(action).manager_ref
+        ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ConfigurationScript.update_in_provider_queue(tower.id, params, auth_user)
+      else
+        job_template = job_template(action) rescue nil
```

Comment:
I left that there for discussion - The `job_template` method is `find_by!` which purposely raises if no record is found - The challenge though is the `nil.configuration_script` error for actually finding the object if the `resource_action` actually exists. I don't want to dumb down the `job_template` method too much - I debated simply adding logic here to do the lookup of the `resource_action` and then handle if that object is `nil`